WordPressAuthenticatorTests/GoogleSignIn/URL+GoogleSignInTests.swift

```diff
@@ -0,0 +1,78 @@
+@testable import WordPressAuthenticator
+import XCTest
+
+class URLGoogleSignInTests: XCTestCase {
+
+    func testGoogleSignInAuthURL() throws {
+        let url = try URL.googleSignInAuthURL(clientId: "123-abc245def.apps.googleusercontent.com")
+
+        assert(url, matchesBaseURL: "https://accounts.google.com/o/oauth2/v2/auth")
+        assertQueryItems(
+            for: url,
+            includeItemNamed: "client_id",
+            withValue: "123-abc245def.apps.googleusercontent.com"
+        )
+        assertQueryItems(
+            for: url,
+            includeItemNamed: "redirect_uri",
+            withValue: "com.googleusercontent.apps.123-abc245def:/oauth2callback"
+        )
+        assertQueryItems(for: url, includeItemNamed: "response_type", withValue: "code")
+        // TODO: need to check more parameters
+    }
+}
+
+func assert(
+    _ actual: URL,
+    matchesBaseURL baseURLString: String,
+    file: StaticString = #file,
+    line: UInt = #line
+) {
+    guard var components = URLComponents(url: actual, resolvingAgainstBaseURL: false) else {
+        return XCTFail(
+            "Could not created `URLComponents` from given `URL` \(actual).",
+            file: file,
+            line: line
+        )
+    }
+
+    components.query = .none
+
+    guard let baseURL = components.url else {
+        return XCTFail(
+            "Could not extract `URL` from `URLComponents` created from \(actual).",
+            file: file,
+            line: line
+        )
+    }
+
+    XCTAssertEqual(baseURL.absoluteString, baseURLString, file: file, line: line)
+}
```

Comment:
This and the other assertion method could go in a dedicated file, I guess.

I might do that if I'll end up writing more tests that need them?